ingresspipe/synapse/examples/store_usage.py

```diff
@@ -6,19 +6,21 @@
 
 from ingresspipe.utils.config_utils import load_yaml
 
-from definitions import CONFIG_PATH, DATA_PATH
+from definitions import ROOT_DIR, CONFIG_PATH, DATA_PATH
 
 config_data = load_yaml(CONFIG_PATH)
 
+PATH_TO_SYN_CONF = os.path.join(ROOT_DIR, '.synapseConfig')
```

Comment:
Also as an aside, as @xindiguo mentioned let's indicate that users shouldn't commit/track the .synapseConfig file. I was wondering if we could remove the .synapseConfig file and instead add instructions on how to generate it using the synapse client (as part of the schematic installation instructions). E.g. something along the lines of (but more refined + example :)
1.  run synapse client login (e.g. using client, cli) with remember me set to true
2.  ensure that PATH_TO_SYN_CONF points to the .synapseConfig file created by the Synapse client 

In fact I wonder, if we could go through step 1 as part of an interactive setup; e.g. if the python synapse client *is not* installed on the user's system already, install the synapse client; ask the user for their username and password; run the synapse client login function w/ the user-provided credentials and remember me set to true - this generates and stores a .synapseConfig file; set the PATH_TO_SYN_CONF to the generated .synapseConfig; if the python synapse client *is* installed on the user's system, either call the script that @xindiguo will provide to find the .synapseConfig file or ask the user to set the PATH_TO_SYN_CONF to point to their existing .synapseConfig. The comment above about including PATH_TO_SYN_CONFIG in definitions/yaml is still valid - that may allow the user to configure which .synapseConfig is used more easily.

Reply:
[Here](https://github.com/Sage-Bionetworks/annotator/blob/64bea145d52343fe4f9391d0699ffcc0f1fe4110/annotator/__main__.py#L22) is one of the examples that checks for login credential and ask for input if the user does not have the config file. If you search `login` in the Python repos, you might find more. I recall there was one looking for the config file first.

Reply:
Also, you can use `synapseclient.client.CONFIG_FILE` as the default path to find the config file

Reply:
@milen-sage Yes, you're right. It does make sense to allow the users to configure the path to their `.synapseConfig` file.

However, the `definitions.py` file is reserved for constants (inspired by Django), so we can allow users to configure the custom path to their `.synapseConfig` file in `config.yml`. If there is a path there we'll parse it out and set the constant in `definitions.py` to that path, else it will default to what we have right now.

I'll save all of this for a different PR though, we should be good to merge the changes on here for now.